db/migrate/20150612135546_create_host_status.rb

```diff
@@ -0,0 +1,36 @@
+class CreateHostStatus < ActiveRecord::Migration
+  def up
+    create_table :host_status do |t|
+      t.string :type
+      t.integer :status, :default => 0, :null => false
+      t.references :host, :null => false
+      t.datetime :reported_at, :null => false
+    end
+    add_index :host_status, :host_id
+    add_foreign_key "host_status", "hosts", :name => "host_status_hosts_host_id_fk", :column => 'host_id'
+    add_column :hosts, :global_status, :integer, :default => 0, :null => false
+
+    Host.all.each do |host|
```

Comment:
Host.all will iterate only over Host::Managed I think, but this feature seems to be on Host::Base.  Should it loop over the latter?